### PR TITLE
DOP-1946: Add sitemap index

### PR DESCRIPTION
**Pull request by sophstad**
[[DOP-1946](https://jira.mongodb.org/browse/DOP-1946)] [[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/landing/sophstad/DOP-1946/sitemap-index.xml)] Add sitemap index for legacy and next-gen sites.

Note that legacy sitemaps are written to `/<project>/master/sitemap.xml.gz` while next-gen projects are written to `/<project>/<current/stable/whatever>/sitemap.xml`, so as projects are converted to next-gen we should be sure to update the URL here.

### CODE REVIEW

**Review by schmalliso — APPROVED** [review]
DOIT.

File: static/sitemap-index.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <loc>https://docs.mongodb.com/guides/sitemap.xml.gz</loc>
+    </sitemap>
+    <sitemap>
+      <loc>https://docs.mongodb.com/kubernetes-operator/master/sitemap.xml.gz</loc>
+    </sitemap>
+    <sitemap>
+      <loc>https://docs.mongodb.com/kafka-connector/current/sitemap.xml</loc>
+    </sitemap>
+    <sitemap>
+      <loc>https://docs.mongodb.com/master/sitemap.xml.gz</loc>
```
Comment (schmalliso) [inline static/sitemap-index.xml:37]: WAIT, should this be master or manual?

Comment (schmalliso) [reply static/sitemap-index.xml:37]: I think master can sometimes == upcoming, rather than stable.

Comment (sophstad) [reply static/sitemap-index.xml:37]: Good catch! I thought that legacy tooling was only outputting sitemaps to `/master` but it looks like there's a sitemap here too: https://docs.mongodb.com/manual/sitemap.xml.gz. Will change!

**Review by schmalliso — APPROVED** [review]
👍🏻

**Pull request merged**